CLI/core/compute_features.py

```diff
@@ -29,98 +20,54 @@
-           "-w", runsolver_watch_data_path,
-           extractor_path / gv.sparkle_extractor_wrapper,
-           "-extractor_dir", extractor_path,
-           "-instance_file"] + instance_list + ["-output_file", result_path]
-    start_time = time.time()
 
-    task_run_status_path = f"Tmp/SBATCH_Extractor_Jobs/{key_str}.statusinfo"
-    status_info_str = (
-        "Status: Running\nExtractor: "
-        f"{extractor_path.name}\n"
-        f"Instance: {instance_name}\n")
-    status_info_str += (
-        "Start Time: "
-        f"{time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(start_time))}\n")
-    status_info_str += "Start Timestamp: " + str(start_time) + "\n"
-    cutoff_str = f"Cutoff Time: {cutoff_time_each_extractor_run} second(s)\n"
-    status_info_str += cutoff_str
-    cmd_str = " ".join([str(c) for c in cmd])
-    subprocess.run(cmd, capture_output=True)
-    end_time = time.time()
+    extractor = Extractor(extractor_path, gv.runsolver_path, gv.sparkle_tmp_path)
+    # We are not interested in the runsolver log, but create the file to filter it
+    # from the extractor call output
+    runsolver_watch_path =\
+        gv.sparkle_tmp_path / f"{instance_path.name}_{extractor_path.name}.wlog"
+    features = extractor.run(instance_list,
+                             feature_group=args.feature_group,
+                             runsolver_args=["--cpu-limit", cutoff_extractor,
+                                             "-w", runsolver_watch_path])
 
     # Now that we have our result, we write it to the FeatureDataCSV with a FileLock
     lock = FileLock(f"{feature_data_csv_path}.lock")
-    try:
-        tmp_fdcsv = sfdcsv.SparkleFeatureDataCSV(result_path,
-                                                 gv.extractor_list)
-        # rename row
-        if not has_instance_set:
-            tmp_fdcsv.dataframe = tmp_fdcsv.dataframe.set_axis(
-                [str(instance_path)], axis=0)
-        else:
-            tmp_fdcsv.dataframe = tmp_fdcsv.dataframe.set_axis(
-                [instance_set.directory / instance_name], axis=0)
+    if features is not None:
         with lock.acquire(timeout=60):
-            feature_data_csv = sfdcsv.SparkleFeatureDataCSV(feature_data_csv_path)
-            feature_data_csv.combine(tmp_fdcsv)
-            feature_data_csv.save_csv()
-        result_string = "Successful"
-    except Exception as ex:
-        print(f"EXCEPTION during retrieving extractor results: {ex}")
-        print(f"****** WARNING: Feature vector computation on instance {instance_path}"
+            feature_data = FeatureDataFrame(feature_data_csv_path)
+            for feature_group, feature_name, value in features:
+                feature_data.set_value(str(instance_name), str(extractor_path),
+                                       feature_group, feature_name, float(value))
+            feature_data.save_csv()
+        lock.release()
+    else:
+        print("EXCEPTION during retrieving extractor results.\n"
```

Comment:
Failure in feature extraction of one feature group is quite common. Maybe we can omit this message to avoid long error messages for "normal" scenarios?

Reply:
In general, this is only show in single jobs .err/.out logs. So you won't see this message N times in a row in a single log. Does that change your view on the matter or should we still omit it?

Reply:
Also, this log only occurs when nothing was extracted, e.g. the extractor completely crashed for a single group. If the extractor reports error values for single features, this error message does not occur.

Reply:
If it is just in single jobs output then it's ok